### PR TITLE
[UR][DeviceSantizer] Enable Symoblizer for UR santizer layer

### PR DESCRIPTION
**Pull request by zhaomaosu**
UR Part: https://github.com/oneapi-src/unified-runtime/pull/1844

### CODE REVIEW

**Comment by omarahmed1111** [comment]
I moved this to one PR that will merge multiple UR PRs together in one bump [here](https://github.com/intel/llvm/pull/14824).

**Comment by omarahmed1111** [comment]
Ping! @intel/dpcpp-sanitizers-review

**Review by AllanZyne — APPROVED** [review]

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -99,14 +101,8 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
...
 
-  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
-  # commit 529e8b9c34acd231eb5b829f6af8123e5f09c974
-  # Merge: 6b33a1bb 5dc794cb
-  # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
-  # Date:   Wed Jul 10 15:05:31 2024 +0100
-  #     Merge pull request #1816 from nrspruit/l0_intel_driver_version
-  #     [L0] Use Intel Level Zero Driver String extension
-  set(UNIFIED_RUNTIME_TAG 529e8b9c34acd231eb5b829f6af8123e5f09c974)
+  set(UNIFIED_RUNTIME_REPO "https://github.com/zhaomaosu/unified-runtime.git")
```
Comment (dm-vodopyanov) [inline sycl/plugins/unified_runtime/CMakeLists.txt:104]: Shouldn't be a fork, correct?

Comment (pbalcer) [reply sycl/plugins/unified_runtime/CMakeLists.txt:104]: correct, this PR needs to wait until https://github.com/intel/llvm/pull/14444 is merged, which has the required UR commits. This PR will then only have the test changes.

**Comment by pbalcer** [comment]
@zhaomaosu the UR side of this change was just merged. You will need to update this PR so that it includes only your test  changes.

**Comment by zhaomaosu** [comment]
> the UR side of this change was just merged. You will need to update this PR so that it includes only your test changes.

sure, updated.

**Review by pbalcer — APPROVED** [review]

**Comment by zhaomaosu** [comment]
Hi @intel/llvm-gatekeepers, this PR is ready to be merged. Thanks.

**Comment by AlexeySachkov** [comment]
It seems that this PR broke post-commit: https://github.com/intel/llvm/actions/runs/10196353773

```
2024-08-01T10:13:43.5810506Z FAILED: lib/libur_loader.so.0.10.0 
2024-08-01T10:13:43.5849875Z : && /opt/sycl/bin/clang++ -fPIC -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wno-covered-switch-default -Wno-error -O3 -DNDEBUG  -Wl,-z,defs -Wl,-z,nodelete   -Xlinker -z -Xlinker relro -Xlinker -z -Xlinker now -Wl,--version-script=/__w/llvm/llvm/build/_deps/unified-runtime-build/source/loader/ur_loader.map -shared -Wl,-soname,libur_loader.so.0 -o lib/libur_loader.so.0.10.0 _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/ur_loader.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/ur_ldrddi.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/ur_libapi.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/ur_libddi.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/ur_lib.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/ur_print.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/validation/ur_valddi.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/validation/ur_validation_layer.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/tracing/ur_tracing_layer.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/tracing/ur_trcddi.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/__/__/__/unified-runtime-src/source/ur/ur.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/sanitizer/asan_allocator.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/sanitizer/asan_buffer.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/sanitizer/asan_interceptor.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/sanitizer/asan_quarantine.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/sanitizer/asan_report.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/sanitizer/asan_shadow_setup.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/sanitizer/stacktrace.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/sanitizer/ur_sanddi.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/sanitizer/ur_sanitizer_layer.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/sanitizer/ur_sanitizer_utils.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/sanitizer/linux/backtrace.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/sanitizer/linux/sanitizer_utils.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/sanitizer/linux/symbolizer.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/validation/backtrace_lin.cpp.o _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/linux/adapter_search.cpp.o  -Wl,-rpath,/__w/llvm/llvm/build/lib:  -lstdc++fs  lib/libur_common.a  lib/libxpti.a  lib/libLLVMSymbolize.so.19.0git  -lstdc++fs  -ldl  -Wl,-rpath-link,/__w/llvm/llvm/build/lib && :
2024-08-01T10:13:43.5861910Z /usr/bin/ld: _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/sanitizer/linux/symbolizer.cpp.o: in function `SymbolizeCode':
2024-08-01T10:13:43.5862697Z symbolizer.cpp:(.text.SymbolizeCode+0x47): undefined reference to `vtable for llvm::raw_string_ostream'
2024-08-01T10:13:43.5863434Z /usr/bin/ld: symbolizer.cpp:(.text.SymbolizeCode+0x68): undefined reference to `llvm::raw_ostream::SetBufferAndMode(char*, unsigned long, llvm::raw_ostream::BufferKind)'
2024-08-01T10:13:43.5864207Z /usr/bin/ld: symbolizer.cpp:(.text.SymbolizeCode+0x158): undefined reference to `llvm::raw_ostream::~raw_ostream()'
2024-08-01T10:13:43.5864814Z /usr/bin/ld: symbolizer.cpp:(.text.SymbolizeCode+0x24e): undefined reference to `llvm::raw_ostream::~raw_ostream()'
2024-08-01T10:13:43.5865402Z /usr/bin/ld: symbolizer.cpp:(.text.SymbolizeCode+0x265): undefined reference to `llvm::raw_ostream::~raw_ostream()'
2024-08-01T10:13:43.5866425Z /usr/bin/ld: _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/sanitizer/linux/symbolizer.cpp.o: in function `void llvm::function_ref<void (llvm::ErrorInfoBase const&, llvm::StringRef)>::callback_fn<SymbolizeCode::$_0>(long, llvm::ErrorInfoBase const&, llvm::StringRef)':
2024-08-01T10:13:43.5867717Z symbolizer.cpp:(.text._ZN4llvm12function_refIFvRKNS_13ErrorInfoBaseENS_9StringRefEEE11callback_fnIZ13SymbolizeCodeE3$_0EEvlS3_S4_+0x2b): undefined reference to `llvm::raw_ostream::write(char const*, unsigned long)'
2024-08-01T10:13:43.5868781Z /usr/bin/ld: symbolizer.cpp:(.text._ZN4llvm12function_refIFvRKNS_13ErrorInfoBaseENS_9StringRefEEE11callback_fnIZ13SymbolizeCodeE3$_0EEvlS3_S4_+0x89): undefined reference to `llvm::raw_ostream::write(unsigned char)'
2024-08-01T10:13:43.5869890Z /usr/bin/ld: _deps/unified-runtime-build/source/loader/CMakeFiles/ur_loader.dir/layers/sanitizer/linux/symbolizer.cpp.o:(.data._ZN4llvm30VerifyDisableABIBreakingChecksE+0x0): undefined reference to `llvm::DisableABIBreakingChecks'
2024-08-01T10:13:43.5870799Z clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```

This is a self-build with shared libraries and assertions disabled

**Comment by sarnex** [comment]
If @zhaomaosu is not available, can someone from @intel/unified-runtime-reviewers take a look? Should we revert this? If I don't hear anything in about an hour I'll revert it since build is broken.

**Comment by kbenzie** [comment]
> If @zhaomaosu is not available, can someone from @intel/unified-runtime-reviewers take a look? Should we revert this? If I don't hear anything in about an hour I'll revert it since build is broken.

I've got very little idea about the sanitizer layer change I'm afraid. It'll probably need to be reverted to unblock things. I believe @zhaomaosu timezone means they are currently asleep (I may be wrong).

**Comment by sarnex** [comment]
@kbenzie I'll watch it and let you know.

**Comment by sarnex** [comment]
@kbenzie Linux and Mac build passed, and Windows build passed even before the commit was reverted, so I think we are all set.

**Comment by kbenzie** [comment]
thanks @sarnex

**Comment by zhaomaosu** [comment]
Thank you all. Sorry for the inconvenience, I'll investigate the failures and reland the changes.

**Pull request merged**